.changeset/twelve-moles-appear.md

```diff
@@ -0,0 +1,5 @@
+---
+'sigstore': minor
+---
+
+Expose new `retry` and `timeout` options to control fetch behavior when errors occure
```

Comment:
Just looking at the documentation for [make-fetch-happen retry](https://github.com/npm/make-fetch-happen#opts-retry) and kinda feeling like we're tying our interface to the underlying http lib which might change in future. 

Thoughts on just exposing `retries: number` and `timeout: number`? We could then have freedom to tune the defaults that make sense for the sigstore public good, as we'll have good insight into which errors will be retriable, what the timeouts/backoffs should be etc.

Reply:
Yeah, I agree it is a bit "leaky" (that isn't even a `make-fetch-happen` thing, it's really the interface for the underlying `retry` library). The reason I opted to go with this approach is that this is also the same interface that the npm CLI uses internally (and exposes through some [command line args](https://github.com/npm/cli/blob/f06469607b80faf72eb897ec3e33deebc6dc10fc/lib/utils/config/definitions.js#L654-L709)).  The public interface for `npm-registry-fetch` also exposes the same [retry option](https://github.com/npm/npm-registry-fetch#-optsretry).

Since our primary use case has us sandwiched between npm CLI and `make-fetch-happen` it seemed easiest to just adopt the data structure which those two had already agreed upon -- that way, any future changes will just flow through `sigstore-js` without us having to do anything.

Reply:
> Since our primary use case has us sandwiched between npm CLI and make-fetch-happen it seemed easiest to just adopt the data structure which those two had already agreed upon -- that way, any future changes will just flow through sigstore-js without us having to do anything.

Yeah these are good point, being tightly integrated in npm cli and will probably continue to be the main use-case for the foreseeable. Part of me still leans toward the simple option and ignoring all the other retry options but 🤷